07.ls_object/lib/screen.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'io/console/size'
+
+class Screen
+  def initialize(src_data, options = {})
+    @src_data = src_data
+
+    @long_format = options[:long_format]
+    @reverse = options[:reverse]
+    @header = options[:header]
+
+    _, @console_width = IO.console_size
+  end
+
+  def show
+    output_blocks = []
+
+    output_blocks << create_output_block_with_file_stats
+    output_blocks << create_output_blocks_with_dir_stats
+
+    puts output_blocks.reject(&:empty?).join("\n\n")
+  end
+
+  private
+
+  def create_output_block_with_file_stats
+    @long_format ? show_detail(@src_data['']) : show_normal(@src_data[''])
+  end
+
+  def create_output_blocks_with_dir_stats
+    @src_data.except('').map do |dir_path, stats|
+      block = []
+
+      block << "#{dir_path}:" if @header
+      block << "total #{stats.map(&:blocks).sum}" if @long_format
+
+      block << (@long_format ? show_detail(stats, dir_path) : show_normal(stats, dir_path))
```

Comment:
```suggestion
      block << (@long_format ? show_detail(stats, dir_path) : show_normal(stats, dir_path))
```

totalの表示もshow_detailの中でやれば条件分岐が減らせます